src/l500/l500-device.cpp

```diff
@@ -643,6 +636,124 @@ namespace librealsense
+            rv = *_temperatures;
+        }
+        else
+        {
+            auto fw_version_support_nest = (_fw_version >= firmware_version("1.5.0.0")) ? true : false;
+            auto expected_size = fw_version_support_nest ? sizeof(extended_temperatures)
+                : sizeof(temperatures);
+
+
+            const auto res = _hw_monitor->send(command{ TEMPERATURES_GET });
+            // Verify read
+            if (res.size() < expected_size)
+            {
+                throw std::runtime_error(
+                    to_string() << "TEMPERATURES_GET - Invalid result size!, expected: "
+                    << expected_size << " bytes, "
+                    "got: " << res.size() << " bytes");
+            }
+
+            auto t = reinterpret_cast<temperatures*>((void*)res.data());
+            rv.LDD_temperature = t->LDD_temperature;
+            rv.MC_temperature = t->MC_temperature;
+            rv.MA_temperature = t->MA_temperature;
+            rv.APD_temperature = t->APD_temperature;
+            rv.HUM_temperature = t->HUM_temperature;
+            rv.AlgoTermalLddAvg_temperature = t->AlgoTermalLddAvg_temperature;
+
+            if (fw_version_support_nest)
+            {
+                extended_temperatures *t_ex = reinterpret_cast<extended_temperatures*>((void*)res.data());
+                rv.nest_avg = t_ex->nest_avg;
+            }
+
+        }
+        return rv;
+    }
+
+    void l500_device::start_temperatures_fetcher()
+    {
+        LOG_DEBUG("Starting temperature fetcher thread");
+        _temperature_fetcher = std::thread( [&]() {
+            try
+            {
+                auto fw_version_support_nest = ( _fw_version >= firmware_version( "1.5.0.0" ) ) ? true : false;
+                auto expected_size = fw_version_support_nest ? sizeof( extended_temperatures )
+                                                             : sizeof( temperatures );
+
+                utilities::time::periodic_timer pt(std::chrono::seconds(1));
+                pt.set_expired(); // Force condition true on start
+
+                _temperature_fetcher_on = true;
+
+                while (_temperature_fetcher_on)
+                {
+                    if (pt) // Update temperatures every second
+                    {
+                        const auto res = _hw_monitor->send(command{ TEMPERATURES_GET });
+                        // Verify read
+                        if (res.size() < expected_size)
+                        {
+                            throw std::runtime_error(
+                                to_string() << "TEMPERATURES_GET - Invalid result size!, expected: "
+                                << expected_size << " bytes, "
+                                "got: " << res.size() << " bytes");
+                        }
+               
+                        std::lock_guard<std::mutex> lock(_temperature_mutex);
+                        auto t = reinterpret_cast<temperatures*>((void*)res.data());
```

Comment:
```
memset( &_temperatures, sizeof( _temperatures ), 0 );
if( fw_version_support_nest )
    _temperatures = *reinterpret_cast< extended_temperatures const * >( res.data() );
else
    *reinterpret_cast< temperatures * >( &_temperatures ) = *reinterpret_cast< temperatures const * >( res.data() );
```
?

Reply:
Even better -- add operator=() to `extended_temperatures`

Reply:
> ```
> memset( &_temperatures, sizeof( _temperatures ), 0 );
> if( fw_version_support_nest )
>     _temperatures = *reinterpret_cast< extended_temperatures const * >( res.data() );
> else
>     *reinterpret_cast< temperatures * >( &_temperatures ) = *reinterpret_cast< temperatures const * >( res.data() );
> ```
> 
> ?

Suggestion taken